cmd/pinniped-proxy/src/service.rs

```diff
@@ -83,7 +83,12 @@ pub async fn proxy(mut req: Request<Body>, default_ca_data: Vec<u8>) -> Result<R
     match client.request(req).await {
         Ok(r) => {
             info!("{}", logging::response_log_data(&r, log_data));
-            Ok(r)
+            if r.status() == StatusCode::SWITCHING_PROTOCOLS {
+                // TODO(agamez): implement ws proxing here; see https://github.com/kubeapps/kubeapps/issues/2328
+                Ok(Response::builder().status(StatusCode::NOT_IMPLEMENTED).body(Body::from("pinniped-proxy does not support websockets yet")).unwrap())
```

Comment:
Great, thanks Antonio. I guess we could do this response without even needing to try proxying the request, but totally fine to do here based on the proxied-response.

Reply:
Yes, this situation can be retrieved from the request as stated by the [RFC 6455](https://tools.ietf.org/html/rfc6455#section-1.2).; it is just a matter of detecting, for example, a header `Connection: Upgrade`. However, it was a little bit easier for me just to compare using the status code from the response.